app.go

```diff
@@ -87,7 +101,15 @@ func (s *App) Provide(constructors ...interface{}) {
 //
 // See dig.Invoke for moreinformation.
 func (s *App) Start(ctx context.Context, funcs ...interface{}) error {
-	return withTimeout(ctx, func() error { return s.start(funcs...) })
+	if s.state != stopped {
+		s.logger.Panic(errors.New("Start can not be called if the App is already running"))
+	}
+	s.state = starting
+	if err := withTimeout(ctx, func() error { return s.start(funcs...) }); err != nil {
```

Comment:
hmm, so if there is an error during start(e.g. timeout), next time we'll retry the start will panic, because the state is not stopped?